be/src/storage/update_manager.cpp

```diff
@@ -504,6 +504,7 @@ Status UpdateManager::on_rowset_finished(Tablet* tablet, Rowset* rowset) {
 
     Status st;
 
+    std::lock_guard lg(*tablet->updates()->get_drop_lock());
```

Comment:
this lock may hold for a long time(index->load), blocking tablet drop, is there any other approach?

Reply:
> this lock may hold for a long time(index->load), blocking tablet drop, is there any other approach?

Remove the `_drop_lock` and we just try to remove index entry from cache when drop tablet. If remove failed, the index entry could be removed by evict thread.